src/components/molecules/TableHeader/TableHeader.tsx

```diff
@@ -149,62 +134,59 @@ const TableHeader: React.FC<TableHeaderProps> = ({
     };
   }, [leaveSeat]);
 
-  const updateTableTitle = useCallback(
-    async (newTableTitle) => {
+  const [updateResponse, updateTables] = useAsyncFn(
+    async (values: EditTableForm) => await updateTableInfo(values),
+    []
+  );
+
+  const updateTableInfo = useCallback(
+    async (values: EditTableForm) => {
       if (!user) return;
 
       const venueTables = tables.map((table) => {
         if (table.reference === tableOfUser?.reference) {
-          return { ...table, title: newTableTitle };
+          return {
+            ...table,
+            title: values.title,
+            subtitle: values.description,
+            capacity: values.capacity,
+          };
         }
         return table;
       });
```

Comment:
What's the intent of this? To update just one specific table? This seems more like an implementation detail that could be hidden away at the `api/` layer and/or backend function or similar rather than being done at the component level like this.

Reply:
Ahhhh, this is a tricky one. Some venues don't have `config.tables` and they fallback to `generateTables` (this generates tables and shows them statically) because of this, I have to iterate the tables, update the current one and then save the whole thing in the database. Yes, all of this can be moved on `api` level but a new backend function needs to be added.

Reply:
At a minimum let's move this to `api/*` layer, but new backend func for it is the ideal situation.

Reply:
Moving this to the `api/*` layer with a transaction will force me to add another rule for `match /venues/{venue}` and allow users to publicly update venues. Which is not a great idea. Maybe we should stick with the current logic/implementation.

Reply:
How is moving the function from one file to another going to change how the security rules work?

Reply:
I moved this to the `api` layer but it's still using the same backend function. I will write another one as you already suggested in one of your previous comments.